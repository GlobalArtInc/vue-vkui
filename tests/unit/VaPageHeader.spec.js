import { mount, shallowMount } from '@vue/test-utils'
import VaPageHeader from '@/PageHeader/VaPageHeader.vue'

describe('VaPageHeader.vue', () => {
  it('renders correctly', () => {
    const wrapper = mount(VaPageHeader)
    expect(wrapper.element).toMatchSnapshot()
  })

  it('does not render empty breadcrumb wrapper', () => {
    const wrapper = shallowMount(VaPageHeader, {
      slots: {}
    })

    expect(wrapper.find('.va-page-header-breadcrumb-container').exists()).toBeFalsy()
  })

  it('renders breadcrumb wrapper', () => {
    const wrapper = shallowMount(VaPageHeader, {
      slots: {
        breadcrumb: '<strong>Test</strong>'
      }
    })

    expect(wrapper.find('.va-page-header-breadcrumb-container').exists()).toBeTruthy()
  })

  it('does not render empty bottom wrapper', () => {
    const wrapper = shallowMount(VaPageHeader, {
      slots: {}
    })

    expect(wrapper.find('.va-page-header-bottom-wrapper').exists()).toBeFalsy()
  })

  it('renders bottom wrapper', () => {
    const wrapper = shallowMount(VaPageHeader, {
      slots: {
        bottom: '<strong>Test</strong>'
      }
    })

    expect(wrapper.find('.va-page-header-bottom-wrapper').exists()).toBeTruthy()
  })

  it('does not render empty actions wrapper', () => {
    const wrapper = shallowMount(VaPageHeader, {
      slots: {}
    })

    expect(wrapper.find('.va-page-header-actions-wrapper').exists()).toBeFalsy()
  })

  it('renders actions wrapper', () => {
    const wrapper = shallowMount(VaPageHeader, {
      slots: {
        actions: '<strong>Test</strong>'
      }
    })

    expect(wrapper.find('.va-page-header-actions-wrapper').exists()).toBeTruthy()
  })

  it('does not render empty subtitle wrapper', () => {
    const wrapper = shallowMount(VaPageHeader, {
      slots: {}
    })

    expect(wrapper.find('h2').exists()).toBeFalsy()
  })

  it('renders subtitle wrapper', () => {
    const wrapper = shallowMount(VaPageHeader, {
      slots: {
        subtitle: '<strong>Test</strong>'
      }
    })

    expect(wrapper.find('h2').exists()).toBeTruthy()
  })
})
