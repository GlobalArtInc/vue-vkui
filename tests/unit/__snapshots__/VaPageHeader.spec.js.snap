// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VaPageHeader.vue renders correctly 1`] = `
<div
  class="va-page-header"
>
  <!---->
   
  <div
    class="va-page-header-title-wrapper"
  >
    <div
      class="va-page-header-title-container"
    >
      <h1 />
       
      <!---->
    </div>
     
    <!---->
  </div>
   
  <!---->
</div>
`;
